controllers/shifts.js

```diff
@@ -60,15 +60,31 @@ module.exports = {
       console.log(err)
     }
   },
+
+  // Triggering condition: only if availability.length == 0 && ownerDisplay == false
   deleteShift: async (req, res) => {
     try {
       await Shift.findOneAndDelete({ _id: req.body.shiftID })
       console.log('Shift deleted')
-      res.json('Shift deleted')
+
     } catch(err) {
     console.log(err)
     }
   },
+  putCafeDisplayFalse: async (req, res) => {
+    try {
+      const shiftData = await Shift.findOneAndUpdate( {_id: req.body.shiftID }, {
+        ownerDisplay: false
+      }, { new: true })
+      if( shiftData.availability.length == 0 ) {
+        module.exports.deleteShift(req)

```

Comment:
You might consider moving this logic out of the controller (do I sound like a broken record?). You could use a post-save middleware to delete a shift after it's saved if certain conditions are met. https://mongoosejs.com/docs/middleware.html#post

Think of your models as describing the "business domain" of your application. As you write your project, you are describing the domain of connecting Cafes and Baristas. You've decided that, when a cafe sets `ownerDisplay` to `false`, and no Baristas have made themselves available for it, that shift should be deleted. This is part of the domain that your project describes. That kind of decision is best described on the model. It allows you to set `ownerDisplay` to `false` somewhere else in your code, and have the model handle the side effects of that if necessary.